libbeat/publisher/queue/memqueue/eventloop.go

```diff
@@ -307,29 +260,46 @@ func newBufferingEventLoop(b *broker, size int, minEvents int, flushTimeout time
 }
 
 func (l *bufferingEventLoop) run() {
-	var (
-		broker = l.broker
-	)
+	broker := l.broker
 
 	for {
+		var pushChan chan pushRequest
```

Comment:
Is this duplicated entirely between here and `newDirectEventLoop`? It is hard for me to spot if there is some subtle difference between the two just scrolling up and down.